R/html-tweak.R

```diff
@@ -571,6 +570,58 @@ tweak_404 <- function(html, pkg = pkg) {
   TRUE
 }
 
+# Add syntax highlighting to reference sections
+tweak_reference_topic_html <- function(html, pkg = pkg) {
+  # blocks with language information = r
+  r_blocks <- xml2::xml_find_all(html, "//div[contains(@class, 'sourceCode r')]/pre/code")
```

Comment:
Maybe this would be simpler if you did something like:

```R
code_blocks <- xml2::xml_find_all(html, "//div/pre/code")
div_class <- code_blocks %>%
  xml2::xml_parent() %>%
  xml2::xml_parent() %>%
  xml2::xml_attr("class")
```

And then made some logical vectors to select from `code_blocks`? 
```

Reply:
You mean simpler than the three ways to identify the three type of code blocks?
From what I can see the problem is that the number of parents is not the same for blocks with language information vs blocks without language information as roxygen2 only adds the div with class "sourceCode someLanguage" in the presence of language information cf https://github.com/r-lib/roxygen2/blob/1cc294fd4bcf989bba647cc603fe826eed566059/R/markdown.R#L259.